src/App.tsx

```diff
@@ -1,25 +1,258 @@
-import React from 'react';
-import logo from './logo.svg';
-import './App.css';
+import React, { useState } from "react";
+import Button from "@mui/material/Button";
+import Typography from "@mui/material/Typography";
+import AppBar from "@mui/material/AppBar";
+import Grid from "@mui/material/Grid";
+import { Card } from "./component/Card";
+import { useAppStyles } from "./App.style";
 
-function App() {
+const CARD_DATA: CardType[] = [
+  { id: 1, design: "♣️", isFlipped: false, isMatched: false },
+  { id: 2, design: "❤️", isFlipped: false, isMatched: false },
+  { id: 3, design: "♦️", isFlipped: false, isMatched: false },
+  { id: 4, design: "♣️", isFlipped: false, isMatched: false },
+  { id: 5, design: "♠️", isFlipped: false, isMatched: false },
+  { id: 6, design: "♦️", isFlipped: false, isMatched: false },
+  { id: 7, design: "♠️", isFlipped: false, isMatched: false },
+  { id: 8, design: "❤️", isFlipped: false, isMatched: false },
+  { id: 9, design: "😂", isFlipped: false, isMatched: false },
+  { id: 10, design: "😂", isFlipped: false, isMatched: false },
+  { id: 11, design: "🦋", isFlipped: false, isMatched: false },
+  { id: 12, design: "🦋", isFlipped: false, isMatched: false },
+  { id: 13, design: "🐶", isFlipped: false, isMatched: false },
+  { id: 14, design: "🐶", isFlipped: false, isMatched: false },
+  { id: 15, design: "⭐️", isFlipped: false, isMatched: false },
+  { id: 16, design: "⭐️", isFlipped: false, isMatched: false },
+];
+
+export type CardType = {
+  id: number;
+  design: string;
+  isFlipped: boolean;
+  isMatched: boolean;
+};
+
+function App(): JSX.Element {
+  const [cards, setCards] = useState<Array<CardType>>([]);
+  const [moves, setMoves] = useState<number>(0);
+
+  const matchedCards = cards.filter((card) => card.isMatched);
+
+  const shuffleArray = (array: CardType[]) => {
+    for (let i = array.length - 1; i > 0; i--) {
+      let j = Math.floor(Math.random() * (i + 1));
+      [array[i], array[j]] = [array[j], array[i]];
+    }
+    return array;
+  };
+
+  const handleCardClick = (clickedCard: CardType) => {
+    const updatedCards = [...cards];
+    const cardIndex = cards.findIndex((card) => card.id === clickedCard.id);
+
+    updatedCards[cardIndex] = {
+      ...updatedCards[cardIndex],
+      isFlipped: true,
+    };
+    setCards(updatedCards);
```

Comment:
We are using setCards here and in the `if` block below as well. In case the if condition is true, we will be setting cards twice in the same function which should not be the case. I don't think we need this setCards in the first place. Please fix this.